src/TSTransformer/util/spreadDestruction/spreadDestructSet.ts

```diff
@@ -0,0 +1,38 @@
+import luau from "@roblox-ts/luau-ast";
+import { TransformState } from "TSTransformer/classes/TransformState";
+
+export function spreadDestructSet(
+	state: TransformState,
+	parentId: luau.AnyIdentifier,
+	index: number,
+	idStack: Array<luau.AnyIdentifier>,
+) {
+	const extracted = state.pushToVar(luau.set(idStack), "extracted");
+	const rest = state.pushToVar(luau.array(), "rest");
+	const keyId = luau.tempId("k");
+
+	state.prereq(
+		luau.create(luau.SyntaxKind.ForStatement, {
+			ids: luau.list.make(keyId, luau.tempId("v")),
```

Comment:
```suggestion
			ids: luau.list.make(keyId),
```
Don't need the unused tempId